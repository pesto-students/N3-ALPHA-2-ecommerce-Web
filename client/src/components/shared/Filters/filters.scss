@import '../../shared/Style/index';

.filters{
    width: 268px;
    min-height: 200px;
    background-color:$cardBackground;
    padding: 20px 40px;

    &_title{
        font-size: $titleSize;
        font-weight: bold;
        margin: 0px;
    }
}