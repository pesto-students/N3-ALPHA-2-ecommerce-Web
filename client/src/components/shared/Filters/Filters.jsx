import { Slider } from 'antd';
import './filters.scss';
import { useTranslation } from 'react-i18next';

export default function Filters(props) {
    const marks = {
        0: '₹0.99',
        199: '₹199',
    };

    const handlePriceChange = (range) => {
        props.onFiltersChange({ price: range });
    };
    const { t } = useTranslation();

    return (
        <div className="filters">
            <p className="filters_title" align="left">
                {t('price_text')}
            </p>
            <Slider
                onChange={handlePriceChange}
                min={0.99}
                max={199}
                range
                marks={marks}
                defaultValue={[0.99, 199]}
            />
        </div>
    );
}
