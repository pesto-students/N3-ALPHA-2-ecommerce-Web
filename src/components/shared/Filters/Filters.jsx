import { Slider } from 'antd';
import './filters.scss';

export default function Filters(props) {
    const marks = {
        0: '₹0.99',
        199: '₹199',
    };

    const handlePriceChange = (range) => {
        props.onFiltersChange({ price: range });
    };
    return (
        <div className="filters">
            <p className="filters_title" align="left">
                Price range
            </p>
            <Slider
                onChange={handlePriceChange}
                min={0.99}
                max={199}
                range
                marks={marks}
                defaultValue={[0.99, 199]}
            />
        </div>
    );
}
