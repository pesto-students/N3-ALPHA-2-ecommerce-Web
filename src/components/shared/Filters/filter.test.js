import renderer from 'react-test-renderer';
import Filters from './Filters';

test('Filter Snapshot', () => {
    const tree = renderer.create(<Filters />).toJSON();
    expect(tree).toMatchSnapshot();
});
