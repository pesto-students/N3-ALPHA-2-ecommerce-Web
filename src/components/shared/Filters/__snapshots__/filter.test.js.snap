// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Snapshot 1`] = `
<div
  className="filters"
>
  <p
    align="left"
    className="filters_title"
  >
    price_text
  </p>
  <div
    className="ant-slider ant-slider-with-marks"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="ant-slider-rail"
      style={Object {}}
    />
    <div
      className="ant-slider-track ant-slider-track-1"
      style={
        Object {
          "left": "0.00505024998737438%",
          "right": "auto",
          "width": "99.99494975001262%",
        }
      }
    />
    <div
      className="ant-slider-step"
    >
      <span
        className="ant-slider-dot"
        style={
          Object {
            "left": "0.49997474875006315%",
          }
        }
      />
      <span
        className="ant-slider-dot ant-slider-dot-active"
        style={
          Object {
            "left": "100%",
          }
        }
      />
    </div>
    <div
      aria-disabled={false}
      aria-valuemax={199}
      aria-valuemin={0.99}
      aria-valuenow={1}
      className="ant-slider-handle ant-slider-handle-1"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="slider"
      style={
        Object {
          "left": "0.00505024998737438%",
          "right": "auto",
          "transform": "translateX(-50%)",
        }
      }
      tabIndex={0}
    />
    <div
      aria-disabled={false}
      aria-valuemax={199}
      aria-valuemin={0.99}
      aria-valuenow={199}
      className="ant-slider-handle ant-slider-handle-2"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="slider"
      style={
        Object {
          "left": "100%",
          "right": "auto",
          "transform": "translateX(-50%)",
        }
      }
      tabIndex={0}
    />
    <div
      className="ant-slider-mark"
    >
      <span
        className="ant-slider-mark-text"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "-0.49997474875006315%",
            "msTransform": "translateX(-50%)",
            "transform": "translateX(-50%)",
          }
        }
      >
        ₹0.99
      </span>
      <span
        className="ant-slider-mark-text ant-slider-mark-text-active"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "100%",
            "msTransform": "translateX(-50%)",
            "transform": "translateX(-50%)",
          }
        }
      >
        ₹199
      </span>
    </div>
  </div>
</div>
`;
